azurerm/resource_arm_storage_account.go

```diff
@@ -548,6 +556,19 @@ func resourceArmStorageAccountCreate(d *schema.ResourceData, meta interface{}) e
 	log.Printf("[INFO] storage account %q ID: %q", storageAccountName, *account.ID)
 	d.SetId(*account.ID)
 
+	enableAdvanceThreatProtection := d.Get("enable_advanced_threat_protection").(bool)
+
+	advancedThreatProtectionSetting := security.AdvancedThreatProtectionSetting{
+		AdvancedThreatProtectionProperties: &security.AdvancedThreatProtectionProperties{
+			IsEnabled: &enableAdvanceThreatProtection,
+		},
+	}
+
+	_, err = advancedThreatProtectionClient.Create(ctx, d.Id(), advancedThreatProtectionSetting)
+	if err != nil {
```

Comment:
These lines should be merged:
```suggestion
	if _, err = advancedThreatProtectionClient.Create(ctx, d.Id(), advancedThreatProtectionSetting); err != nil {
```

Reply:
done